### DISCUSSION TITLE
[Social Cards] Are percentage values for a Layer's x and y position available?

### DISCUSSION THREAD

**Discussion by Andre601** [squidfunk/mkdocs-material#5485] [Q&A] [answered]
I would like to know if a layer allows percentage values instead of full numbers (pixels) to be defined for the X and Y position (Maybe even with an anchor point? More on that a bit later).

If this isn't possible yet should it be considered, as it could be a really tedious task, adjusting a layer's position after changing the card's width and height.
Like if you resize it are you required to also adjust the X and Y position of the layers afterwards to keep the same gaps as you had before.

Additionally am I wondering if layers can have an anchor point set. With anchor point do I mean the point from wich the position/distance is calculated for the layer.
As an example, setting the anchor point to `top-right` would use the top-right corner of the layer to go `n` pixels on the X and Y axis to set it.
This would allow you to have as an example `80%` set for both X and Y (Assuming percentage is supported) without fearing that after a size adjustment, the layer is clipping outside the full canvas, having parts cut of... I hope you get my explanation here.

**Comment by squidfunk** [comment, marked as answer]
Thanks for the feedback! Great to have some user feedback so early, so we can steer the plugin into a good direction. Percentages are not (yet?) supported. Why would you want to change the cards width and height? 1200x630 is the standard for social images.

> Additionally am I wondering if layers can have an anchor point set. With anchor point do I mean the point from wich the position/distance is calculated for the layer.

We will soon add the possibility to set an origin for offset computations.

**Reply by Andre601** [reply]
Truth be told, I don't use the plugin... because no insider and stuff.

But I would be surprised if there wasn't at least one person who may want to have a resolution alongside fullHD or even more...

**Reply by squidfunk** [reply]
As of [insiders+4.35.0](https://squidfunk.github.io/mkdocs-material/insiders/changelog/#4.35.0), the social plugin now supports [setting `origin` for a layer](https://squidfunk.github.io/mkdocs-material/setup/setting-up-social-cards/#origin).

``` yaml
size: { width: 1200, height: 630 }
layers:
  - size: { width: 64, height: 64 }
    offset: { x: 32, y: 32 }
    origin: end bottom
```

This will render a layer `64x64px` in size with a right and bottom spacing of `32px`. Additionally, the plugin will now validate the syntax of the custom layout and catch errors without crashing when the documentation is served (by using `mkdocs serve`), which greatly improves the editing experience. When building, errors crash the build.
